docs/debug.md

```diff
@@ -0,0 +1,650 @@
+The most important log file to look at is:
+
+    /var/log/syslog
+
+The syslog is particularly useful since kernel logs, launcher output, service
+status, system logs and confinement violations (these are covered in depth
+elsewhere) all get logged there.
+
+Further, systemd has log dump facility to look at all system and snap
+services. The "snappy service" command provides an easy way to monitor and
+inspect snap services. To get the status of all snap services run "snappy
+service status". To inspect the logs run "snappy service logs". Both commands
+can take a snap name to limit to a specific snap and also `<snapname>` to
+limit to only a specific service of a snap (ie for the shout service in
+`shout.sergiusens` run: `sudo snappy service logs shout`).
+
+
+## Debugging binaries
+
+### Installing binaries
+In order to test a new snap on a Snappy system you need to install it first.
+This is called sideloading and it can be done via
+`snappy-remote --url=ssh://ubuntu@webdm.local/ install snapname.snap`.
+Alternatively the snap can be copied via scp into the Snappy system and
+installed via `sudo snappy install snapname.snap`.
+
+Note that you have to use the `--allow-unauthenticated` tag for installing
+unsigned snaps: `sudo snappy install --allow-unauthenticated snapname.snap`.
+
+### Binary names on CLI
+
+The name of the binaries will be prefixed with the snap package name. This
+ensures that there are never any namespace collision in the binary names. For
+example, the snap `pastebinit.mvo` contains a `pastebinit` script. The binary
+on disk will be called `pastebinit.pastebinit`. The exception of this rule are
+frameworks which may have short binary names. E.g. the docker package contains
+a docker binary.
+
+### Find a binary in the file system hierarchy
+
+All binary names can be found in `/apps/bin/`. The snappy tool will generate
+small wrapper script that ensures that the binary in the snap is called with
+the right confinement and environment.
+
+### Testing if a binary is running
+After a snap is installed the binaries are available as
+`<snapname.binaryname>` on the commandline. It can be tested by simply
+running it from the commandline. Common issues are that the application tries
+to read/write outside of its confinement. This will result in permission
+denied errors even if the app runs as root. To see if this is the case, the
+`dmesg | tail` command is helpful. The errors are of the form:
+
+```
+[ 8020.798544] audit: type=1400 audit(1442568421.022:9): apparmor="DENIED" operation="open" profile="pastebinit.mvo_pastebinit_1.4.0.0.2" name="/etc/fstab" pid=1123 comm="pastebinit" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
+```
+
+To find out what paths are available to write the `hello-world` package is
+helpful. After installing it the command `hello-world.env` is available that
+will show the environment that the snap binaries see. The
+`SNAP_APP_USER_DATA_PATH` and `SNAP_APP_DATA_PATH` contain the directories
```

Comment:
Yeah try grepping for only SNAP. For more information see the [PR that added them](https://github.com/ubuntu-core/snappy/pull/296).